src/main/groovy/wooga/gradle/unity/utils/internal/UnityAssetFile.groovy

```diff
@@ -19,57 +19,147 @@ package wooga.gradle.unity.utils.internal
 
 import org.gradle.api.logging.Logger
 import org.gradle.api.logging.Logging
+import org.yaml.snakeyaml.DumperOptions
 import org.yaml.snakeyaml.Yaml
+import org.yaml.snakeyaml.nodes.Tag
+import org.yaml.snakeyaml.representer.Represent
+import org.yaml.snakeyaml.representer.Representer
+import org.yaml.snakeyaml.nodes.Node;
 
 abstract class UnityAssetFile {
     static Logger logger = Logging.getLogger(UnityAssetFile)
     private final validObject
+    private final File assetFile
 
     private Map<String, Object> content = [:]
+    private List<String> unityInstructions
 
     UnityAssetFile(File assetFile) {
         this(assetFile.text)
+        this.assetFile = assetFile
     }
 
     UnityAssetFile(String content) {
         boolean isValid = false
         try {
             Yaml parser = new Yaml()
-            Map<String, Object> c = parser.load(stripUnityInstructions(content))
+            def lines = content.readLines()
+            lines.removeAll({ it -> it.isEmpty()})
+            unityInstructions = lines[0,1,2]
+            lines[1] = ""
+            lines[2] = "--- "
+            String stripped = lines.join("\n")
+            Map<String, Object> c = parser.load(stripped)
             this.content = c[c.keySet().first()] as Map<String, Object>
             isValid = true
 
         }
         catch (Exception ignored) {
-            logger.warn("Project Settings file not parsable. Please make sure it's not set to binary.")
+            logger.warn("UnityAssetFile content not parsable. Please make sure it's not set to binary.")
         }
         validObject = isValid
     }
 
+    @Override
+    int hashCode() {
+        return content.hashCode()
+    }
+
+    @Override
+    boolean equals(Object obj) {
+        if (!obj) {
+            return false
+        }
+        if (obj instanceof UnityAssetFile) {
+            UnityAssetFile other = (UnityAssetFile)obj
+            return this.content == other.content
+        }
+        return false
+    }
+
+    Boolean isSerialized() {
+        assetFile != null
+    }
+
     Boolean isValid() {
         validObject
     }
 
+    File getAssetFile(){
+        assetFile
+    }
+
     Map<String, Object> getContent() {
         content
     }
 
-    static String stripUnityInstructions(String content) {
-        def lines = content.readLines()
-        lines.collect {
-            if (it.matches(/%TAG !u! tag:unity3d.com,.*:/)) {
-                return ""
-            }
+    boolean saveToFile() {
```

Comment:
I think the term `flush` or `write` fits better here

Reply:
agreed!